src/OldRod.Core/Disassembly/Inference/InferenceDisassembler.cs

```diff
@@ -252,7 +261,22 @@ private bool ContinueDisassembly(VMFunction function, IEnumerable<ProgramState>
                     // Offset is not visited yet, read instruction. 
                     _decoder.ReaderOffset = (uint) currentState.IP;
                     _decoder.CurrentKey = currentState.Key;
-                    instruction = _decoder.ReadNextInstruction();
+
+                    if (function.SMCTrampolineOffsetRange.IsEmpty && function.BlockHeaders.Contains((long)currentState.IP)
```

Comment:
Ah I suppose this is a perf, to skip SMC trampoline block detection in the middle of a basic block.

Regardless, this should probably be documented so that it doesn't get accidentally removed in the future.

Reply:
This check was indeed added as a performance improvement. We don't need to check every instruction whether or not they are the starting instruction of the SMC trampoline block.